From f7947e9a0e6ae99e8fd0ab3c9176885f724ac87d Mon Sep 17 00:00:00 2001
From: Prabhu Sundararaj <prabhu.sundararaj@nxp.com>
Date: Tue, 19 Jul 2016 12:14:26 -0500
Subject: [PATCH] Avoid attaching NULL buffer while hiding EGL windows

Incremental fix for QTBUG-46921. Destroying the callback for NULL surface is not
working all the time. If the window is hidden by destroying the EGL surface
and hence setting the NULL buffer is no longer required.

Upstream-Status: Pending. Posted for review

Task-number:QTBUG-47902
Signed-off-by: Prabhu Sundararaj <prabhu.sundararaj@nxp.com>
---
 src/client/qwaylandwindow.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index 2dd58e8..24b5653 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -312,8 +312,6 @@ void QWaylandWindow::setVisible(bool visible)
         QPointer<QWaylandWindow> deleteGuard(this);
         QWindowSystemInterface::flushWindowSystemEvents();
         if (!deleteGuard.isNull()) {
-            attach(static_cast<QWaylandBuffer *>(0), 0, 0);
-            commit();
             if (mBackingStore) {
                 mBackingStore->hidden();
             }
-- 
2.7.4

